File: migrate.py
import sys

def extract_body(filepath, output_astro_path, title):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Find start
        start_idx = content.find('<section')
        if start_idx == -1:
            start_idx = content.find('<div id="reading-progress"></div>')
            if start_idx != -1:
                start_idx += len('<div id="reading-progress"></div>')
        
        # Find end
        end_idx = content.find('<div id="dynamic-footer">')
        if end_idx == -1:
            end_idx = content.find('<footer')
        
        if start_idx == -1 or end_idx == -1:
            print(f"Could not find boundaries for {filepath}")
            return
            
        body = content[start_idx:end_idx].strip()
        
        # Remove dynamic-navbar if exists
        nav_str = '<div id="dynamic-navbar"></div>'
        if nav_str in body:
            body = body.replace(nav_str, '')
            
        script_idx = content.find('<script>')
        script_end_idx = content.find('</script>', script_idx)
        script = ""
        if script_idx != -1 and script_end_idx != -1:
            script = content[script_idx:script_end_idx+9]
            # Replace <script> with <script is:inline>
            script = script.replace('<script>', '<script is:inline>')
            
        astro_content = f\"\"\"---
import Layout from '../layouts/Layout.astro';
---
<Layout title="{title}">
  {body}
  
  {script}
</Layout>
\"\"\"
        with open(output_astro_path, 'w', encoding='utf-8') as f:
            f.write(astro_content)
        print(f"Successfully created {output_astro_path}")
    except Exception as e:
        print(f"Error: {e}")

extract_body('pages/about.html', 'src/pages/about.astro', 'Sobre mí | Marketing Cloud Arcade')
extract_body('pages/catalog.html', 'src/pages/catalog.astro', 'Catálogo | Marketing Cloud Arcade')

File: migrate.js
const fs = require('fs');

function extractBody(filepath, outpath, title) {
    const content = fs.readFileSync(filepath, 'utf8');
    let startIdx = content.indexOf('<section');
    if (startIdx === -1) {
        startIdx = content.indexOf('<div id="reading-progress"></div>');
        if (startIdx !== -1) startIdx += 33;
    }
    let endIdx = content.indexOf('<div id="dynamic-footer">');
    if (endIdx === -1) endIdx = content.indexOf('<footer');
    if (startIdx === -1 || endIdx === -1) return console.log('Failed to find boundaries for ' + filepath);
    
    let body = content.substring(startIdx, endIdx).trim();
    body = body.replace('<div id="dynamic-navbar"></div>', '');
    
    let script = '';
    const scriptIdx = content.indexOf('<script>');
    if (scriptIdx !== -1) {
        const endScript = content.indexOf('</script>', scriptIdx);
        if (endScript !== -1) {
            script = content.substring(scriptIdx, endScript + 9).replace('<script>', '<script is:inline>');
        }
    }
    
    const astro = ---
import Layout from '../layouts/Layout.astro';
---
<Layout title=" + title + ">
   + body + 
  
   + script + 
</Layout>;

    fs.writeFileSync(outpath, astro, 'utf8');
    console.log('Created ' + outpath);
}

extractBody('pages/about.html', 'src/pages/about.astro', 'Sobre mí | Marketing Cloud Arcade');
extractBody('pages/catalog.html', 'src/pages/catalog.astro', 'Catálogo | Marketing Cloud Arcade');
